src/controllers/auth.js

```diff
@@ -1,11 +1,94 @@
+const jwt = require('jsonwebtoken');
+
+const bcrypt = require('bcrypt');
+
+const saltRounds = 10;
+
+const User = require('../models/user');
+
 module.exports = {
-  signInUser: (req, res) => {
-    res.send('You successfully signed in');
+  signInUser: async (req, res) => {
+    res.json('');
+    /*
+    const { email, password, rememberMe } = req.body;
+
+    const user = await User.findOne({ email });
+    if (!user) return res.status(400).send('Wrong username or password');
+
+    bcrypt.compare(password, user.password_hash, function (err, result) {
+      if (result) {
+        const payload = {
+          username: username,
+          email: email,
+          exp: (Date.now() + 12096e5) / 1000,
+          iat: Date.now() / 1000,
+        };
+
+        const token = jwt.sign(payload, process.env.JWT_SECRET);
+
+        res.cookie('token', token, {
+          httpOnly: true,
+        });
+        res.status(302).header('user', user.id).redirect('/user/authenticated');
+      } else {
+        res.status(400).send('Wrong username or password');
+      }
+    });
+    */
   },
-  signUpUser: (req, res) => {
-    res.send('You successfully signed up');
+  // eslint-disable-next-line consistent-return
+  signUpUser: async (req, res) => {
+    // res.json('');
+    try {
+      const {
+        firstName,
+        lastName,
+        username,
+        email,
+        password,
+        password2,
+        address,
+        acceptTos,
+      } = req.body;
+
+      if (await User.exists({ username })) {
+        return res.status(400).send('Username already used');
+      }
+      if (password !== password2) {
+        return res.status(400).send('passwords do not match');
+      }
+
+      const passwordHash = await bcrypt.hash(password, saltRounds);
+      const user = await User.create({
+        firstName,
+        lastName,
+        username,
+        email,
+        password_hash: passwordHash,
+        acceptTos,
+        address,
+      });
+
+      const payload = {
+        username,
+        email,
+        exp: (Date.now() + 12096e5) / 1000,
+        iat: Date.now() / 1000,
+      };
+
+      const token = jwt.sign(payload, process.env.JWT_SECRET);
+
+      res.cookie('token', token, {
+        httpOnly: true,
+      });
+      res.json(user);
+    } catch (err) {
+      res.status(422).json({ message: err.message ?? err });
+    }
   },
   signOutUser: (req, res) => {
-    res.send('You successfully signed out');
+    // res.json('');
+
+    res.clearCookie('token');
   },
 };
```

Comment:
Was this auth file merged from another branch? Then I won't review it in this PR.

Reply:
Oh okay, then can you create an issue for setting up auth routes and link it to this PR?
I'll do another review of the auth code as well.

Reply:
I guess I did it, but I am not sure if I did it correctly

Reply:
Yes it's done! You can confirm by seeing at the bottom of the PR activities and also on the right hand side it shows that you linked the issue.

Can you update your PR description to "Closes #13 and #26"?

Reply:
Thank you, Done